### PR TITLE
Fixes #10460: rudder-reports doesn't start postgresql on postinstall on sles 12

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/10460

### CODE REVIEW

**Comment by peckpeck** [comment]
Commit modified

File: rudder-reports/SPECS/rudder-reports.spec [hunk 1]
```diff
@@ -105,33 +105,42 @@ install -m 644 %{SOURCE3} %{buildroot}/opt/rudder/etc/server-roles.d/
...
-/bin/systemctl status  ${POSTGRESQL_SERVICE_NAME}.service
+%if 0%{?rhel} >= 7 || 0%{?suse_version} >= 1200
+  /bin/systemctl status ${POSTGRESQL_SERVICE_NAME}.service
 %endif
 
-if [ $? -ne 0 ]
-then
-%if 0%{?rhel} || 0%{?fedora}
+if [ $? -ne 0 ]; then
+  echo -n "INFO: Initializing PostgreSQL ..."
```
Comment (amousset) [inline rudder-reports/SPECS/rudder-reports.spec]: This line should go in the if {rhel}

**Comment by peckpeck** [comment]
Commit modified

File: rudder-agent/SOURCES/rudder.init [hunk 2]
```diff
@@ -41,9 +41,11 @@ get_service_name()
 ## Service name
   if type systemctl >/dev/null 2>&1; then
     SERVICE_NAME=$(systemctl list-unit-files --type service | awk -F'.' '{print $1}' | grep -E "^${1}[0-9]*$" | tail -n 1)
-  elif type chkconfig >/dev/null 2>&1; then
+  fi
+  if [ -z "${POSTGRESQL_SERVICE_NAME}" ] && type chkconfig >/dev/null 2>&1; then
```
Comment (amousset) [inline rudder-agent/SOURCES/rudder.init]: SERVICE_NAME

File: rudder-agent/SOURCES/rudder.init [hunk 3]
```diff
@@ -41,9 +41,11 @@ get_service_name()
...
     SERVICE_NAME=$(chkconfig 2>/dev/null | awk '{ print $1 }' | grep "${1}" | tail -n 1)
-  else
+  fi
+  if [ -z "${POSTGRESQL_SERVICE_NAME}" ]
```
Comment (amousset) [inline rudder-agent/SOURCES/rudder.init]: same here

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**